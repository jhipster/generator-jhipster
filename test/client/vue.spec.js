const expect = require('expect');
const path = require('path');
const { skipPrettierHelpers: helpers } = require('../utils/utils');
const { OAUTH2 } = require('../../jdl/jhipster/authentication-types');
const { MICROSERVICE } = require('../../jdl/jhipster/application-types');
const { VUE } = require('../../jdl/jhipster/client-framework-types');

const commonOptions = { clientFramework: VUE };

describe('JHipster vue generator', () => {
  describe('microfrontend', () => {
    let runResult;
    before(async () => {
      runResult = await helpers
        .create(path.join(__dirname, '../../generators/client'))
        .withOptions({
          applicationType: MICROSERVICE,
          skipInstall: true,
          auth: OAUTH2,
          enableTranslation: true,
          nativeLanguage: 'en',
          languages: ['fr', 'en'],
          ...commonOptions,
        })
        .run();
    });
    it('should match generated files snapshot', () => {
      expect(runResult.getStateSnapshot()).toMatchSnapshot();
    });
  });
});
