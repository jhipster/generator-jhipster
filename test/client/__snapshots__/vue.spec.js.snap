// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JHipster vue generator microfrontend should match generated files snapshot 1`] = `
Object {
  ".editorconfig": Object {
    "stateCleared": "modified",
  },
  ".eslintignore": Object {
    "stateCleared": "modified",
  },
  ".eslintrc.js": Object {
    "stateCleared": "modified",
  },
  ".gitattributes": Object {
    "stateCleared": "modified",
  },
  ".gitignore": Object {
    "stateCleared": "modified",
  },
  ".husky/pre-commit": Object {
    "stateCleared": "modified",
  },
  ".lintstagedrc.js": Object {
    "stateCleared": "modified",
  },
  ".postcssrc.js": Object {
    "stateCleared": "modified",
  },
  ".prettierignore": Object {
    "stateCleared": "modified",
  },
  ".prettierrc": Object {
    "stateCleared": "modified",
  },
  ".yo-rc.json": Object {
    "stateCleared": "modified",
  },
  "README.md": Object {
    "stateCleared": "modified",
  },
  "package.json": Object {
    "stateCleared": "modified",
  },
  "sonar-project.properties": Object {
    "stateCleared": "modified",
  },
  "src/main/resources/i18n/messages_en.properties": Object {
    "stateCleared": "modified",
  },
  "src/main/resources/i18n/messages_fr.properties": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/404.html": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/WEB-INF/web.xml": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/account/account.service.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/account/login.service.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/configuration/configuration.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/configuration/configuration.service.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/configuration/configuration.vue": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/docs/docs.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/docs/docs.vue": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/health/health-modal.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/health/health-modal.vue": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/health/health.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/health/health.service.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/health/health.vue": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/logs/logs.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/logs/logs.service.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/logs/logs.vue": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/metrics/metrics-modal.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/metrics/metrics-modal.vue": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/metrics/metrics.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/metrics/metrics.service.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/admin/metrics/metrics.vue": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/app.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/app.vue": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/constants.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/core/error/error.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/core/error/error.vue": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/core/home/home.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/core/home/home.vue": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/core/jhi-footer/jhi-footer.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/core/jhi-footer/jhi-footer.vue": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/core/jhi-navbar/jhi-navbar.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/core/jhi-navbar/jhi-navbar.vue": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/core/ribbon/ribbon.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/core/ribbon/ribbon.vue": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/declarations.d.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/entities/entities-menu.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/entities/entities-menu.vue": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/entities/entities.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/entities/entities.vue": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/entities/user/user.oauth2.service.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/index.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/locale/translation.service.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/main.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/router/admin.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/router/entities.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/router/index.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/router/pages.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/alert/alert.service.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/config/axios-interceptor.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/config/config-bootstrap-vue.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/config/config.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/config/dayjs.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/config/formatter.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/config/store/account-store.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/config/store/translation-store.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/data/data-utils.service.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/date/filters.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/jhi-item-count.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/jhi-item-count.vue": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/model/user.model.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/security/authority.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/sort/jhi-sort-indicator.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/sort/jhi-sort-indicator.vue": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/sort/sorts.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shims-vue.d.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/css/loading.css": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_0.svg": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_0_head-192.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_0_head-256.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_0_head-384.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_0_head-512.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_1.svg": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_1_head-192.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_1_head-256.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_1_head-384.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_1_head-512.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_2.svg": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_2_head-192.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_2_head-256.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_2_head-384.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_2_head-512.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_3.svg": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_3_head-192.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_3_head-256.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_3_head-384.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_3_head-512.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/logo-jhipster.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/scss/_bootstrap-variables.scss": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/scss/global.scss": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/scss/vendor.scss": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/favicon.ico": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/activate.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/configuration.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/error.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/global.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/health.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/home.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/login.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/logs.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/metrics.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/password.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/register.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/reset.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/sessions.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/settings.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/user-management.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/activate.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/configuration.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/error.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/global.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/health.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/home.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/login.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/logs.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/metrics.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/password.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/register.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/reset.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/sessions.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/settings.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/user-management.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/index.html": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/manifest.webapp": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/robots.txt": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/swagger-ui/dist/images/throbber.gif": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/swagger-ui/index.html": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/jest.conf.js": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/account/account.service.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/account/login.service.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/admin/configuration/configuration.component.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/admin/health/health-modal.component.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/admin/health/health.component.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/admin/health/health.service.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/admin/logs/logs.component.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/admin/metrics/metrics-modal.component.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/admin/metrics/metrics.component.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/core/error/error.component.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/core/home/home.component.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/core/jhi-navbar/jhi-navbar.component.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/core/ribbon/ribbon.component.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/entities/entities-menu.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/microfrontends/entities-menu.component.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/microfrontends/entities-menu.vue": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/microfrontends/entities-router.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/shared/alert/alert.service.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/shared/config/axios-interceptor.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/shared/config/formatter.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/javascript/spec/app/shared/data/data-utils.service.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/test/resources/i18n/messages_en.properties": Object {
    "stateCleared": "modified",
  },
  "src/test/resources/i18n/messages_fr.properties": Object {
    "stateCleared": "modified",
  },
  "tsconfig.json": Object {
    "stateCleared": "modified",
  },
  "tsconfig.spec.json": Object {
    "stateCleared": "modified",
  },
  "webpack/config.js": Object {
    "stateCleared": "modified",
  },
  "webpack/vue.utils.js": Object {
    "stateCleared": "modified",
  },
  "webpack/webpack.common.js": Object {
    "stateCleared": "modified",
  },
  "webpack/webpack.dev.js": Object {
    "stateCleared": "modified",
  },
  "webpack/webpack.microfrontend.js": Object {
    "stateCleared": "modified",
  },
  "webpack/webpack.prod.js": Object {
    "stateCleared": "modified",
  },
}
`;
